### PR TITLE
fix(icon): PD-355 Use currentcolor to set glyph fills

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Updated Icon component such that fill styling now uses currentColor, and thus CSS color rules can be used to style glyphs. 

🎟 _Jira ticket:_ [PD-355](https://jira.mongodb.org/browse/PD-355)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .svgrrc.js [hunk 1]
```diff
@@ -22,9 +20,21 @@ module.exports = {
...
     return template.ast`
 			${imports}
-			import PropTypes from 'prop-types';
+      import PropTypes from 'prop-types';
+      import { css, cx } from '@leafygreen-ui/emotion';
 			
-			const ${componentName} = (${props}) => ${jsx};
+			const ${componentName} = (${props}) => {
+        // Setting className to manually enable fill prop to overwrite the color when the prop is set
+        props.className = cx(props.className, {[css\`color: \${ props.fill }\`]: props.fill});
```
Comment (hswolff) [inline .svgrrc.js]: I'm confused....how are these props used? Like standard React Props? We're overwriting them?

What does the resulting code look like with this change?

Comment (brookescarlett) [reply .svgrrc.js]: Yes these props are used like standard react props. This is how we are making sure that if a consumer passes the component a fill prop, it's being prioritized over css color rules. Then, if no fill prop is passed, we're just taking the className that they passed and determining color that way (if they have set CSS color rules). 

Not sure that I'm clear on the question that you are asking though

Comment (hswolff) [reply .svgrrc.js]: So we're modifying the props being passed in?

What does the resulting code look like? 

In a typical react component you wouldn't modify props this way, is my point. 

```
function Foo(props) {
   props.blah = 'blah';
}
```

Comment (brookescarlett) [reply .svgrrc.js]: So the consumer is interacting with the Icon component, which wraps the SVGComponent. We're changing the way that the SVGComponent reads the props passed to the Icon component.

**Pull request merged**